bin/spack

```diff
@@ -29,10 +29,13 @@ from __future__ import print_function
 import os
 import sys
 
-if sys.version_info[:2] < (2, 6):
+if sys.version_info[:2] < (2, 6) or (
+        sys.version_info[:2] >= (3, 0)
+    and sys.version_info[:2] < (3, 5)
+    ):
     v_info = sys.version_info[:3]
-    sys.exit("Spack requires Python 2.6 or higher."
-             "This is Python %d.%d.%d." % v_info)
+    sys.exit("Spack requires Python 2.6, 2.7 or 3.5-3.9"
```

Comment:
can we just say "3.5 or higher"?  I think it'll make it easier to maintain as we'll just keep bumping the lower bound.

On that note, can you refactor it so that we really only have to change one thing each time we tweak the support?

i.e.:

```python

min_python2 = (2, 6)
min_python3 = (3, 5)

if sys.version_info[:2] < min_python2 or (
        sys.version_info[:2] >= (3, 0)
    and sys.version_info[:2] < min_python3
    ):
    msg = "Spack requires Python 2.6, 2.7 or %d.%d or higher" % min_python3
    msg += "You ran Spack with Python %d.%d.%d" % v_info
    sys.exit(msg)
```

Reply:
TBH that's my preference as well, but I copied the message from wherever `spack debug report` prints the error as a start.  Will update in a moment.